### PR TITLE
Paywalls: Add `PaywallView`

### PR DESCRIPTION
**Pull request by tonidero**
### Description
Full screen counterpart to #1509. This adds a new view `PaywallView` which wraps the `Paywall` composable so it's usable in the view system. 

The view will be used in the hybrids.

### CODE REVIEW

File: ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/views/PaywallView.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                shouldShowCloseButton = if (hasValue(R.styleable.PaywallView_shouldDisplayDismissButton)) {
+                    getBoolean(R.styleable.PaywallView_shouldDisplayDismissButton, false)
+                } else {
+                    null
+                }
+            } finally {
+                recycle()
+            }
+        }
+        fontFamilyId?.let {
```
Comment (vegaro) [inline ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/views/PaywallView.kt:140]:
```suggestion
        fontFamilyId?.takeIf { it > 0 }?.let {
```

**Review by vegaro — APPROVED** [review]
There is some code shared between `PaywallView` and `PaywallFooterView`. I was wondering if it would be worth it to extract it to a common class. For example, the bug with the font family resource id being 0 could have been fixed in both views at the same time if we had that code somewhere else that's common for both. But I am not sure what's the best strategy to share that code.

**Comment by tonidero** [comment]
> There is some code shared between `PaywallView` and `PaywallFooterView`. I was wondering if it would be worth it to extract it to a common class. For example, the bug with the font family resource id being 0 could have been fixed in both views at the same time if we had that code somewhere else that's common for both. But I am not sure what's the best strategy to share that code.

Ohh right... There is one more parameter currently in the full screen view than that footer view, and this uses ids, but I will think of ways to share that, thanks!

**Comment by vegaro** [comment]
@tonidero can you take a look at the merge conflicts resolution?

**Comment by tonidero** [comment]
Took a look, makes sense! :shipit:

**Comment by NachoSoto** [comment]
Awesome!

**Pull request merged**